### ISSUE TITLE
Feature Request: `IgnoredGroups` configuration in `RSpec/NestedGroups`

### ISSUE THREAD

**Issue by jtannas** [rubocop/rubocop-rspec#1360]
TL;DR: Please add `IgnoredGroups` configuration to `RSpec/NestedGroups` so that we can stop configured groups from contributing to the nesting level.

### Context

We use the [`RSwag`](https://github.com/rswag/rswag) gem in our Rails request specs. A hiccup we're running into is it uses a lot of custom example groups for boilerplate descriptions of the API format and it's causing problems with the `RSpec/NestedGroups` cop.

eg.
```ruby
RSpec.describe '/users', type: :request do # <-- Nesting group 1
  path '/users' do # <-- Nesting group 2
    post 'creates a new user' do # <-- Nesting group 3
      response '201', 'Created' do # <-- Nesting group 4
        # tests go here
      end

      response '401', 'Unauthorized' do
        # tests go here
      end
      # etc...
```

We're stuck with either setting a very high level of allowed nesting or disabling the cop.

### Requested Solution
The feature I'd like is to tell Rubocop not to count configured example groups against the nesting level of the cop

```yml
RSpec/NestedGroups:
  IgnoreGroups:
    - path
    - get
    - post
    - delete
    - put
    - response
```
```ruby
RSpec.describe '/users', type: :request do # <-- Cop nesting level += 1
  path '/users' do # <-- Cop nesting level += 0
    post 'creates a new user' do # <-- Cop nesting level += 0
      response '201', 'Created' do # <-- Cop nesting level += 0
        # tests go here
      end
```

Thanks for any feedback and help you can provide :bow: 

---
Relevant Cop Docs: https://www.rubydoc.info/gems/rubocop-rspec/1.7.0/RuboCop/Cop/RSpec/NestedGroups
Relevant Cop Code: https://github.com/rubocop/rubocop-rspec/blob/master/lib/rubocop/cop/rspec/nested_groups.rb
